r/tests/testthat/test-Array.R

```diff
@@ -532,6 +532,9 @@ test_that("Array$create() can handle data frame with custom struct type (not inf
   a <- Array$create(df, type = type)
   expect_type_equal(a$type, type)
 
+  ## as_arrow_array respects `type` argument (ARROW-17620)
+  expect_type_equal(a, as_arrow_array(df, type = type))
```

Comment:
As we have a few other tests for the functionality of `as_arrow_array()`, and this is testing one of the function's arguments, I think it's significant enough to live in its own `test_that()` block.  Would you mind making that change, and moving it down (I think those tests start near line 1021 of this file)?